### PR TITLE
Compatibility with osTicket 1.17

### PR DESCRIPTION
**Pull request by adam-dej**

### CODE REVIEW

File: slack/slack.php [hunk 1]
```diff
@@ -10,11 +10,13 @@ class SlackPlugin extends Plugin {
   var $config_class = "SlackPluginConfig";
 
   function bootstrap() {
-    Signal::connect('ticket.created', array($this, 'onTicketCreated'), 'Ticket');
+    $config = $this->getConfig();
+    Signal::connect('ticket.created', function ($ticket) use($config) {
```
Comment (lubo) [inline slack/slack.php]:
```suggestion
    Signal::connect('ticket.created', function ($ticket) use ($config) {
```

It'd be nice to stay consistent.

**Review by lubo — COMMENTED** [review]
Do you have any reference to the change in `Plugin` behavior?

**Comment by adam-dej** [comment]
> Do you have any reference to the change in `Plugin` behavior?

The change where `$config` is cleared was introduced here: https://github.com/osTicket/osTicket/commit/9f2cc16e3536d772a87f40d0c563cc4840af4584. But even if it wasn't, this plugin wouldn't behave correctly in 1.17 if user enabled more instances of it.

As I understand this, plugin init sequence (https://github.com/osTicket/osTicket/blob/v1.17/include/class.plugin.php#L193) now is:
1. An object is created from the `Plugin` subclass.
2. Active plugin instances are enumerated.
3. For each instance, its `bootstrap` (https://github.com/osTicket/osTicket/blob/v1.17/include/class.plugin.php#L1157) function runs:
  - This function uses `$plugin->getConfig($this)`, which has a side-effect of caching `$config` of the instance in the object created in step 1.
  - This function then calls `bootstrap()` on object created in step 1. Therefore, we can access `$config` of the current instance from the `bootstrap` function we've overridden.
  - repeat (note: object created in step 1. is re-used for bootstraping of other instances, therefore its `$config` is re-written)
4. Set `$config` on object from step 1. to `null`.

**Comment by lubo** [comment]
I see. I'd rephrase the title of 087f504aeeb5f44c2101ae570192535456d53bc2 to something like `Introduce compatibility with osTicket 1.17` and squash 4571daadcf6e802049a9effd295e38a66ef263b6 into it. I think it'll be ready for merge then.

**Comment by adam-dej** [comment]
Done.

**Review by lubo — APPROVED** [review]
